tests/scan.cc

```diff
@@ -286,3 +286,152 @@ TEST(scan_depth_zero) {
+      (void*)&cap_with_perms
+#else
+      cheri_sentry_create(cheri_address_set(cheri_pcc_get(), (ptraddr_t)(void*)&cap_with_perms))
+#endif
+      // open slot for sealcap
+  };
+  size_t sz = sizeof(caps[7]);
+  sysctlbyname("security.cheri.sealcap", &caps[7], &sz, NULL, 0);
+
+  Mapper mapper;
+  mapper.maps()->push_back(std::make_unique<capmap::BasicMap>("store", "virtual memory", PERM_w));
+  mapper.maps()->push_back(
+      std::make_unique<capmap::BasicMap>("store cap", "virtual memory", PERM_wW));
+  mapper.maps()->push_back(
+      std::make_unique<capmap::BasicMap>("load/store", "virtual memory", PERM_r | PERM_w));
+  mapper.maps()->push_back(
+      std::make_unique<capmap::BasicMap>("load/store cap", "virtual memory", PERM_rR | PERM_wW));
+  mapper.maps()->push_back(std::make_unique<capmap::BasicMap>("seal", "otype", CHERI_PERM_SEAL));
+  mapper.maps()->push_back(std::make_unique<capmap::BranchMap>());
+
+  mapper.scan(CAP_TO(caps), "caps");
+
+  if (options().verbose()) {
+    mapper.print_json(stdout);
+  }
+  TRY(mapper.maps()->size() == 6);
+  auto w_map = dynamic_cast<capmap::BasicMap const*>(mapper.maps()->at(0).get());
+  auto wW_map = dynamic_cast<capmap::BasicMap const*>(mapper.maps()->at(1).get());
+  auto rw_map = dynamic_cast<capmap::BasicMap const*>(mapper.maps()->at(2).get());
+  auto rwRW_map = dynamic_cast<capmap::BasicMap const*>(mapper.maps()->at(3).get());
+  auto seal_map = dynamic_cast<capmap::BasicMap const*>(mapper.maps()->at(4).get());
+  auto branch_map = dynamic_cast<capmap::BranchMap const*>(mapper.maps()->at(5).get());
+
+  // construct expected outcomes for each memory map
+  SparseRange w_expect(Range::from_cap(caps[1]));
+  SparseRange wW_expect(Range::from_cap(caps[4]));
+  SparseRange rw_expect(Range::from_cap(caps[2]));
+  SparseRange rwRW_expect(Range::from_cap(caps[5]));
+  rwRW_expect.combine(Range::from_cap(CAP_TO(caps)));
+  rw_expect.combine(rwRW_expect);
+  wW_expect.combine(rwRW_expect);
+  w_expect.combine(rw_expect);
+  w_expect.combine(wW_expect);
+
+  if (options().verbose()) {
+    fprintf(stderr, "w_expect : ");
+    w_expect.print_json(stderr, "\t");
+  }
+
+  TRY(w_map->sparse_range() == w_expect);
+  TRY(wW_map->sparse_range() == wW_expect);
+  TRY(rw_map->sparse_range() == rw_expect);
+  TRY(rwRW_map->sparse_range() == rwRW_expect);
+  TRY(seal_map->sparse_range() == SparseRange(Range::from_cap(caps[7])));
+  TRY(branch_map->sparse_range() ==
+      SparseRange(Range::from_base_length(cheri_address_get(caps[6]), 1)));
+
+#ifdef __CHERI_PURE_CAPABILITY__
+  for (int i = 0; i < 6; i++) free(caps[i]);
+#else
+  for (int i = 0; i < 6; i++) free((void*)cheri_address_get(caps[i]));
+#endif
+}
+
+static int pois_acc = 0;
+bool pois_cb(void* __capability cap) {
+  (void)cap;
+  pois_acc++;
+  return false;
+}
+
+TEST(scan_poison_map) {
+  typedef struct node {
+    struct node* __capability next;
+  } node_t;
+  node_t* __capability early, * __capability poison_node;
```

Comment:
These are better written on separate lines, because the `*` binding is weird.

Reply:
Fixed.

Reply:
@djlowther If you could say "Fixed in <xyz>" where <xyz> is a commit hash, it'll really help your reviewer!

Reply:
Sorry, I'm new to this detailed PR process. Fixed in [efcec74](https://github.com/capablevms/compartment-mapper/pull/3/commits/efcec74bc78a25f735ca16a72654c7afafc4386d)